modules/cluster/variables.tf

```diff
@@ -189,6 +189,30 @@ variable "critical_addons_node_group_bottlerocket_admin_container_source" {
   description = "URI of a custom admin container image"
 }
 
+variable "critical_addons_vpc-cni_configuration_values" {
+  type        = string
+  default     = null
+  description = "Configuration values passed to the vpc-cni EKS addon."
+}
+
+variable "critical_addons_kube-proxy_configuration_values" {
+  type        = string
+  default     = null
+  description = "Configuration values passed to the kube-proxy EKS addon."
+}
+
+variable "critical_addons_coredns_configuration_values" {
+  type        = string
+  default     = null
+  description = "Configuration values passed to the coredns EKS addon."
+}
+
+variable "critical_addons_ebs-csi_configuration_values" {
+  type        = string
+  default     = null
```

Comment:
I had to double-check, but passing `null` through to the `eks_addon` resource is equivalent to not passing the option

https://developer.hashicorp.com/terraform/language/expressions/types#null

Reply:
CI is failing with 

> Message_: "ConfigurationValue provided in request is not supported: Json schema validation failed with error: [$: null found, object expected]"

so we need to check this. Checking the jsonschemas for the addon configurations we can pass an empty object as the default

Reply:
or keep null as default and don't call `jsonencode` in the `aws_eks_addon` resources

Reply:
Ah of course - the `jsonencode` is the source of the problem.

Its not _totally_ clear what the expected behaviour is for `configuration_values = "{}"`, because whilst none of the fields are included in `required`, they also have no defined jsonschema `default`.

Reply:
Ah yes you're right - perhaps better to avoid the call to `jsonencode` then 👍